Objects/object.c

```diff
@@ -2970,3 +2971,51 @@ _Py_SetRefcnt(PyObject *ob, Py_ssize_t refcnt)
 {
     Py_SET_REFCNT(ob, refcnt);
 }
+
+
+static PyObject* constants[] = {
+    &_Py_NoneStruct,                   // Py_NONE id
+    (PyObject*)(&_Py_FalseStruct),     // Py_FALSE id
+    (PyObject*)(&_Py_TrueStruct),      // Py_TRUE id
+    &_Py_EllipsisObject,               // Py_ELLIPSIS id
+    &_Py_NotImplementedStruct,         // Py_NOT_IMPLEMENTED id
+};
+
+PyObject*
+Py_GetConstant(unsigned int constant_id)
+{
+    if (constant_id <= Py_NOT_IMPLEMENTED) {
+        return constants[constant_id];
+    }
+
+    PyObject *obj;
+    switch (constant_id) {
+    case Py_ZERO:
+        obj = _PyLong_GetZero();
+        break;
+    case Py_ONE:
+        obj = _PyLong_GetOne();
+        break;
+    case Py_EMPTY_STR:
+        obj = PyUnicode_New(0, 0);
+        break;
+    case Py_EMPTY_BYTES:
+        obj = PyBytes_FromStringAndSize(NULL, 0);
+        break;
+    case Py_EMPTY_TUPLE:
+        obj = PyTuple_New(0);
+        break;
+    default:
+        return NULL;
+    }
+    return obj;
+}
+
+
+PyObject*
+Py_GetConstantRef(unsigned int constant_id)
+{
+    PyObject *obj = Py_GetConstant(constant_id);
+    assert(obj == NULL || _Py_IsImmortal(obj));
```

Comment:
Maybe two separate asserts? That would be easier to debug if it goes off. :-)

Reply:
Fixed